### ISSUE TITLE
RTX 3090 Cuda 10.1 Kernal image error

### ISSUE THREAD

**Issue by frank-qcd-qk** [tensorflow/tensorflow#43718]
**System information**
RTX3090, AMD 3950x

Nvidia Driver 455.23

Fri Oct  2 00:41:55 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 455.23.05    Driver Version: 455.23.05    CUDA Version: 11.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce RTX 3090    On   | 00000000:0D:00.0  On |                  N/A |
| 30%   33C    P8    31W / 350W |    590MiB / 24265MiB |     24%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1881      G   /usr/lib/xorg/Xorg                350MiB |
|    0   N/A  N/A      2159      G   /usr/bin/gnome-shell               52MiB |
|    0   N/A  N/A      2560      G   ...AAAAAAAAA= --shared-files       47MiB |
|    0   N/A  N/A      2565      G   ...token=9020920654293169663       35MiB |
|    0   N/A  N/A      3061      G   ...mviewer/tv_bin/TeamViewer       35MiB |
|    0   N/A  N/A      5620      G   ...AAAAAAAAA= --shared-files       64MiB |
+-----------------------------------------------------------------------------+

NVCC:
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Sun_Jul_28_19:07:16_PDT_2019
Cuda compilation tools, release 10.1, V10.1.243

Tensorflow version:
v2.3.0-54-gfcc4b966f1 2.3.1

**Describe the current behavior**
<pre>2020-10-02 00:44:35.590969: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
Observed TF Version:  2.3.1
Observed Numpy Version:  1.18.5
Directory already exists!
Observation Length:  28693
Linear Length:  28693
Angular Length:  28693
Load all complete
Currently using multiple GPUs
2020-10-02 00:44:37.843072: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcuda.so.1
2020-10-02 00:44:37.876858: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-10-02 00:44:37.877518: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties: 
pciBusID: 0000:0d:00.0 name: GeForce RTX 3090 computeCapability: 8.6
coreClock: 1.695GHz coreCount: 82 deviceMemorySize: 23.70GiB deviceMemoryBandwidth: 871.81GiB/s
2020-10-02 00:44:37.877531: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2020-10-02 00:44:37.878382: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcublas.so.10
2020-10-02 00:44:37.879285: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcufft.so.10
2020-10-02 00:44:37.879431: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcurand.so.10
2020-10-02 00:44:37.880196: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusolver.so.10
2020-10-02 00:44:37.880602: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusparse.so.10
2020-10-02 00:44:37.882240: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.7
2020-10-02 00:44:37.882327: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-10-02 00:44:37.883011: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-10-02 00:44:37.883625: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1858] Adding visible gpu devices: 0
Selected MultiGPU but only observe single GPU. Cancel Multi GPU training!
2020-10-02 00:44:37.889428: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2020-10-02 00:44:37.894039: I tensorflow/core/platform/profile_utils/cpu_utils.cc:104] CPU Frequency: 3493600000 Hz
2020-10-02 00:44:37.894764: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5f88190 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-10-02 00:44:37.894780: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2020-10-02 00:44:37.954768: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-10-02 00:44:37.955482: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5f6fd90 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2020-10-02 00:44:37.955503: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): GeForce RTX 3090, Compute Capability 8.6
2020-10-02 00:44:37.955670: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-10-02 00:44:37.956825: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties: 
pciBusID: 0000:0d:00.0 name: GeForce RTX 3090 computeCapability: 8.6
coreClock: 1.695GHz coreCount: 82 deviceMemorySize: 23.70GiB deviceMemoryBandwidth: 871.81GiB/s
2020-10-02 00:44:37.956850: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2020-10-02 00:44:37.956877: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcublas.so.10
2020-10-02 00:44:37.956891: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcufft.so.10
2020-10-02 00:44:37.956905: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcurand.so.10
2020-10-02 00:44:37.956918: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusolver.so.10
2020-10-02 00:44:37.956931: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusparse.so.10
2020-10-02 00:44:37.956945: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.7
2020-10-02 00:44:37.957008: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-10-02 00:44:37.958197: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-10-02 00:44:37.959374: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1858] Adding visible gpu devices: 0
2020-10-02 00:44:37.959401: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
</pre>

Halts for a minute then error:

<pre>Traceback (most recent call last):
  File &quot;train.py&quot;, line 94, in &lt;module&gt;
    model = FrankNet.build(200, 150)
  File &quot;/home/frank/Duckietown/AIDO5/challenge-aido_LF-baseline-behavior-cloning/duckieTrainer/frankModel.py&quot;, line 76, in build
    linearVelocity = FrankNet.build_linear_branch(inputs)
  File &quot;/home/frank/Duckietown/AIDO5/challenge-aido_LF-baseline-behavior-cloning/duckieTrainer/frankModel.py&quot;, line 12, in build_linear_branch
    x = Conv2D(24, (5, 5), strides=(2, 2), padding=&quot;valid&quot;)(x)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py&quot;, line 925, in __call__
    return self._functional_construction_call(inputs, args, kwargs,
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py&quot;, line 1098, in _functional_construction_call
    self._maybe_build(inputs)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py&quot;, line 2643, in _maybe_build
    self.build(input_shapes)  # pylint:disable=not-callable
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/keras/layers/convolutional.py&quot;, line 197, in build
    self.kernel = self.add_weight(
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py&quot;, line 597, in add_weight
    variable = self._add_variable_with_custom_getter(
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/training/tracking/base.py&quot;, line 745, in _add_variable_with_custom_getter
    new_variable = getter(
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer_utils.py&quot;, line 133, in make_variable
    return tf_variables.VariableV1(
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/variables.py&quot;, line 260, in __call__
    return cls._variable_v1_call(*args, **kwargs)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/variables.py&quot;, line 206, in _variable_v1_call
    return previous_getter(
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/variables.py&quot;, line 67, in getter
    return captured_getter(captured_previous, **kwargs)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/distribute/distribute_lib.py&quot;, line 2857, in creator
    return next_creator(**kwargs)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/variables.py&quot;, line 199, in &lt;lambda&gt;
    previous_getter = lambda **kwargs: default_variable_creator(None, **kwargs)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/variable_scope.py&quot;, line 2583, in default_variable_creator
    return resource_variable_ops.ResourceVariable(
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/variables.py&quot;, line 264, in __call__
    return super(VariableMetaclass, cls).__call__(*args, **kwargs)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/resource_variable_ops.py&quot;, line 1507, in __init__
    self._init_from_args(
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/resource_variable_ops.py&quot;, line 1651, in _init_from_args
    initial_value() if init_from_fn else initial_value,
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/keras/initializers/initializers_v2.py&quot;, line 397, in __call__
    return super(VarianceScaling, self).__call__(shape, dtype=_get_dtype(dtype))
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/init_ops_v2.py&quot;, line 561, in __call__
    return self._random_generator.random_uniform(shape, -limit, limit, dtype)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/init_ops_v2.py&quot;, line 1043, in random_uniform
    return op(
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/util/dispatch.py&quot;, line 201, in wrapper
    return target(*args, **kwargs)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/ops/random_ops.py&quot;, line 288, in random_uniform
    shape = tensor_util.shape_tensor(shape)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/framework/tensor_util.py&quot;, line 1029, in shape_tensor
    return ops.convert_to_tensor(shape, dtype=dtype, name=&quot;shape&quot;)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/framework/ops.py&quot;, line 1499, in convert_to_tensor
    ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py&quot;, line 338, in _constant_tensor_conversion_function
    return constant(v, dtype=dtype, name=name)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py&quot;, line 263, in constant
    return _constant_impl(value, dtype, shape, name, verify_shape=False,
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py&quot;, line 275, in _constant_impl
    return _constant_eager_impl(ctx, value, dtype, shape, verify_shape)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py&quot;, line 300, in _constant_eager_impl
    t = convert_to_eager_tensor(value, ctx, dtype)
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py&quot;, line 97, in convert_to_eager_tensor
    ctx.ensure_initialized()
  File &quot;/home/frank/.local/lib/python3.8/site-packages/tensorflow/python/eager/context.py&quot;, line 539, in ensure_initialized
    context_handle = pywrap_tfe.TFE_NewContext(opts)
tensorflow.python.framework.errors_impl.InternalError: CUDA runtime implicit initialization on GPU:0 failed. Status: device kernel image is invalid
</pre>

**Describe the expected behavior**
Code should execute fine.

**Comment by frank-qcd-qk** [comment]
Ubuntu version 20.04

**Comment by frank-qcd-qk** [comment]
@mihaimaruseac I noticed that in another thread you mentioned there is a libtensorflow upload, will that potentially also fix this issue?

**Comment by mihaimaruseac** [comment]
Only if this issue stems from libtensorflow, not from the pip package

**Comment by amahendrakar** [comment]
@frank-qcd-qk,
Looking at issue [#41990](https://github.com/tensorflow/tensorflow/issues/41990#issuecomment-686853339) with a similar error log, seems like the issue is fixed in the latest TF-nightly. 

Could you please check if you are facing the same issue with TF-nightly, CUDA 11 and cuDNN 8? Thanks!

**Comment by frank-qcd-qk** [comment]
Is Cuda 10.1 just not going to be supported?

**Comment by mihaimaruseac** [comment]
TF 2.3 (2.2 and 2.1 too) builds against CUDA 10.1. TF 2.4 will build against CUDA 11.

**Comment by frank-qcd-qk** [comment]
> TF 2.3 (2.2 and 2.1 too) builds against CUDA 10.1. TF 2.4 will build against CUDA 11.

Yes, I understand. so does that me whatever issue caused this problem with 30 series graphics card will not be addressed until TF2.4?

**Comment by frank-qcd-qk** [comment]
I will test later today with the latest CUDA 11, and TF 2.4, but it would be nice if we can still use CUDA 10.1 and TF 2.3.

**Comment by frank-qcd-qk** [comment]
> TF 2.3 (2.2 and 2.1 too) builds against CUDA 10.1. TF 2.4 will build against CUDA 11.

Also is it CUDA 11 or 11.1?

**Comment by mihaimaruseac** [comment]
Nightly already builds with CUDA 11. You can test with that, if possible, if you need CUDA 11.

@sanjoy / @pkanwar23 do we build CUDA 11.0 or CUDA 11.1?

**Comment by sanjoy** [comment]
Yeah, unfortunately TF 2.3 does not support GPUs with compute capability 8.x.  tf-nightly and the upcoming TF 2.4 release should both work, PTAL.  Both tf-nightly and TF 2.4 will use CUDA 11.

**Comment by frank-qcd-qk** [comment]
@sanjoy  @mihaimaruseac  I have indeed used tf-nightly and cuda 11.0. I constantly get the issue of:

Internal: ptxas exited with non-zero error code 65280, output: ptxas fatal   : Value 'sm_86' is not defined for option 'gpu-name'

Are we going to support cuda11.1 in nightly? Since it seems Cuda11.0 just don't support 3090?

**Comment by sanjoy** [comment]
> Internal: ptxas exited with non-zero error code 65280, output: ptxas fatal : Value 'sm_86' is not defined for option 'gpu-name'

Can you share the full log?  Wherever this call is coming from, we should be able to ask ptxas to generate SASS for sm_80 since SASS for sm 80 should work for sm 86.

**Comment by frank-qcd-qk** [comment]
@sanjoy From Nvidia Cuda 11.0 does not support SM86. For RTX 30 series card, should use SM86 capability. I have unfortunately nuked the setup and go with a source build. However, the error message is:

2020-09-22 11:42:03.984823: W tensorflow/stream_executor/cuda/redzone_allocator.cc:312]
Internal: ptxas exited with non-zero error code 65280, output: ptxas fatal   : Value 'sm_86' is not defined for option 'gpu-name'

as pointed out by Pudget systems: https://www.pugetsystems.com/labs/hpc/RTX3090-TensorFlow-NAMD-and-HPCG-Performance-on-Linux-Preliminary-1902/

And it seems for using tensorflow that uses tf1 behavior, it will break the code, for tf code 2.0 and above, it seems fine.

**Comment by soluwalana** [comment]
I can confirm that the RTX 3090 works with the following configuration:
tf-nightly-gpu 2.4.0-dev20201016 
Linux version 5.4.0-51-generic (buildd@lcy01-amd64-020) (gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2))
Driver Version: 455.23.05
Cuda Version 11.0.228 (failed to load with 10.1, 10.2, or 11.1)
cuDNN 8.0.3.33 (failed with 7)

**Comment by sanjoy** [comment]
Hi @frank-qcd-qk,

Are you sure you're using a recent build?  The path in your log indicates that the TF build you're using is quite old.  Moreover, the error message you see should not be a fatal error, TF should fall back to the driver JIT.

**Comment by BorhenJlidi** [comment]
> Yeah, unfortunately TF 2.3 does not support GPUs with compute capability 8.x. tf-nightly and the upcoming TF 2.4 release should both work, PTAL. Both tf-nightly and TF 2.4 will use CUDA 11.

I built TF 2.3.1 under CUDA 11.1 in a machine with RTX 3080, I haven't had any errors. Does this mean TF 2.3 doesn't utilize the GPU fully. Could you please elaborate on this?

**Comment by sanjoy** [comment]
> I built TF 2.3.1 under CUDA 11.1 in a machine with RTX 3080, I haven't had any errors.

I was talking about the TF binary pip packages we ship.  If you're building from source you can include support for any compute capability (>= 3.5) you like.

**Comment by BorhenJlidi** [comment]
@sanjoy understood.

**Comment by Harsh188** [comment]
I'm facing the same issue. I recently purchased a RTX 3080 and followed the instructions on [Installing TensorFlow GPU in Ubuntu 20.04](https://towardsdatascience.com/installing-tensorflow-gpu-in-ubuntu-20-04-4ee3ca4cb75d).

I have installed:
- Cuda 10.1
- CuDNN 7.6.5

Which are the recommended drivers on the TensorFlow docs for [GPU support](https://tensorflow.google.cn/install/gpu).
When I run `tf.config.list_physical_devices("GPU")` I get the following output:
`[PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]`

But when I was testing it in jupyter notebook
```python
if tf.test.gpu_device_name():
    print('Default GPU Device:{}'.format(tf.test.gpu_device_name()))
```
It resulted in teh following error:
_**RuntimeError:** CUDA runtime implicit initialization on GPU:0 failed. Status: device kernel image is invalid_

**Useful system information**
Ubuntu 20.04

`$ nvcc -V`
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Sun_Jul_28_19:07:16_PDT_2019
Cuda compilation tools, release 10.1, V10.1.243

`$ nvidia-smi`
![Screenshot from 2020-10-22 18-07-35](https://user-images.githubusercontent.com/47203161/96872673-9060ab80-1491-11eb-8ed8-2eaaaedaba43.png)

I have also opened up #44229 to address the issue of the lack of instructions for ubuntu 20.04

**Comment by BorhenJlidi** [comment]
@Harsh188 , you have two versions of CUDA (10.1 and 11.1). That might have triggered the error, but I sense that this is a driver issue. Could you tell us if you're running TF on your machine or in a container and what's the actual Nvidia driver version that is loaded? In my case, drivers <455 failed to recognize the RTX 3080. And keep in mind that depending on the installation method you have chosen for CUDA 10.1, you might have installed an older bundled driver with the installation.

**Comment by Harsh188** [comment]
@BorhenJlidi I think the default CUDA that it ships with is 11.1. I manually went thru and downloaded 10.1 since that's what TensorFlow recommends. My driver version is 455.28 and I'm running it on my machine. I'm still learning how to use docker.

**Comment by BorhenJlidi** [comment]
@Harsh188 I recommend building TF with CUDA 11.1 and CUDNN v8.0.4.30, but you'd be obliged to turn TRT off for this build as it's not supported yet for CUDA 11.1. I tried different build configurations and this one worked for me. If you specifically need CUDA 10.1 and/or TRT enabled, you should figure out how to keep the 455 driver installed and loaded with breaking things up (you might have to install CUDA 10.1 manually from compressed archive). No matter what version of CUDA you use, a driver older than 455 will cause TF to not recognize you device (RTX 3080). Unless RTX 30 software support is full,  I do not recommend building a  Tensorflow/Tensorflow Serving docker image for RTX 3080 as builds currently fail and TensorRT for CUDA 11.1 isn't out yet.

**Comment by frank-qcd-qk** [comment]
The confirmed workaround is to compile against the latest cuda, cudnn, tensorrt. Confirmed tf-nightly works but not with compute 8.6 capability and will get warning.

**Comment by BorhenJlidi** [comment]
> @Harsh188 I recommend building TF with CUDA 11.1 and CUDNN v8.0.4.30, but you'd be obliged to turn TRT off for this build as it's not supported yet for CUDA 11.1. I tried different build configurations and this one worked for me. If you specifically need CUDA 10.1 and/or TRT enabled, you should figure out how to keep the 455 driver installed and loaded with breaking things up (you might have to install CUDA 10.1 manually from compressed archive). No matter what version of CUDA you use, a driver older than 455 will cause TF to not recognize you device (RTX 3080). Unless RTX 30 software support is full, I do not recommend building a Tensorflow/Tensorflow Serving docker image for RTX 3080 as builds currently fail and TensorRT for CUDA 11.1 isn't out yet.

@Harsh188 @frank-qcd-qk TensorRT-7.2.1.6 is released, you can now build with TensorRT support.

**Comment by Harsh188** [comment]
@BorhenJlidi I'm having trouble setting up my workspace. Can you check out #44307 and help me out? Thanks.

**Comment by wwww666** [comment]
Does RTX3090 adapt to CUDA10.1+Cudnn7.6+tensorflow2.x now? There is also CUDA9.0, because I also need to use tensorflow1.x version

**Comment by longmalongma** [comment]
> I can confirm that the RTX 3090 works with the following configuration:
> tf-nightly-gpu 2.4.0-dev20201016
> Linux version 5.4.0-51-generic (buildd@lcy01-amd64-020) (gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2))
> Driver Version: 455.23.05
> Cuda Version 11.0.228 (failed to load with 10.1, 10.2, or 11.1)
> cuDNN 8.0.3.33 (failed with 7)



> configuration

Thanks, do your configuration compile successfully? Can your configuration run codes needs to be compiled on 3090? looking forward to your reply.

**Comment by soluwalana** [comment]
> > I can confirm that the RTX 3090 works with the following configuration:
> > tf-nightly-gpu 2.4.0-dev20201016
> > Linux version 5.4.0-51-generic (buildd@lcy01-amd64-020) (gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2))
> > Driver Version: 455.23.05
> > Cuda Version 11.0.228 (failed to load with 10.1, 10.2, or 11.1)
> > cuDNN 8.0.3.33 (failed with 7)
> 
> > configuration
> 
> Thanks, do your configuration compile successfully? Can your configuration run codes needs to be compiled on 3090? looking forward to your reply.

Yes this configuration compiles computation graphs and executes them on the 3090

**Comment by longmalongma** [comment]
> > > I can confirm that the RTX 3090 works with the following configuration:
> > > tf-nightly-gpu 2.4.0-dev20201016
> > > Linux version 5.4.0-51-generic (buildd@lcy01-amd64-020) (gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2))
> > > Driver Version: 455.23.05
> > > Cuda Version 11.0.228 (failed to load with 10.1, 10.2, or 11.1)
> > > cuDNN 8.0.3.33 (failed with 7)
> > 
> > 
> > > configuration
> > 
> > 
> > Thanks, do your configuration compile successfully? Can your configuration run codes needs to be compiled on 3090? looking forward to your reply.
> 
> Yes this configuration compiles computation graphs and executes them on the 3090

do you mean compile you can compile c++ codes into .so files, and run it by python?

**Comment by longmalongma** [comment]
> > > > I can confirm that the RTX 3090 works with the following configuration:
> > > > tf-nightly-gpu 2.4.0-dev20201016
> > > > Linux version 5.4.0-51-generic (buildd@lcy01-amd64-020) (gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2))
> > > > Driver Version: 455.23.05
> > > > Cuda Version 11.0.228 (failed to load with 10.1, 10.2, or 11.1)
> > > > cuDNN 8.0.3.33 (failed with 7)
> > > 
> > > 
> > > > configuration
> > > 
> > > 
> > > Thanks, do your configuration compile successfully? Can your configuration run codes needs to be compiled on 3090? looking forward to your reply.
> > 
> > 
> > Yes this configuration compiles computation graphs and executes them on the 3090
> 
> do you mean compile you can compile c++ codes into .so files, and run it by python?

**Comment by Amokstakov** [comment]
@soluwalana 
I am trying to match your configurations and I not getting any luck. I downloaded the TF-nightly but I cannot download the CUDA version with that driver - any help?

**Comment by Amokstakov** [comment]
> > Yeah, unfortunately TF 2.3 does not support GPUs with compute capability 8.x. tf-nightly and the upcoming TF 2.4 release should both work, PTAL. Both tf-nightly and TF 2.4 will use CUDA 11.
> 
> I built TF 2.3.1 under CUDA 11.1 in a machine with RTX 3080, I haven't had any errors. Does this mean TF 2.3 doesn't utilize the GPU fully. @sanjoy Could you please elaborate on this?

Can you please walk through your steps to reproduce your set-up/downloads for running with 3080? I believe I tried with a whole slew of different versions and I can't seem to get it to work

**Comment by BorhenJlidi** [comment]
@Amokstakov check my reply to another user having you problem https://github.com/tensorflow/tensorflow/issues/44307#issuecomment-716547268 or simply install these packages (please note that the inclusion of extensions deb,tgz.. is intended as I believe picking those yields an easier and more flexible install experience and don't forget to follow the relevant NVIDIA's installation guide by version and installer extension/format ) :

- cuda-repo-ubuntu2004-11-1-local_11.1.0-455.23.05-1_amd64.deb
- cudnn-11.1-linux-x64-v8.0.4.30.tgz
- TensorRT-7.2.1.6.Ubuntu-18.04.x86_64-gnu.cuda-11.1.cudnn8.0.tar.gz

and build Tensorflow 2.3.1 from source and you're good to go!

**Comment by fspider** [comment]
The same issue here.

- RTX 3090
- Cuda 10.1
- CUDNN 7.6.5
- Tensorflow-gpu 2.2.0

isn't there any news with this issue?

**Comment by amahendrakar** [comment]
> isn't there any news with this issue?

@fspider,
Could you please submit a new issue from [this link](https://github.com/tensorflow/tensorflow/issues/new/choose) and fill in the template, so that we can track the issue there. Thanks!

**Comment by frank-qcd-qk** [comment]
Update:
With tensorflow 2.4 now officially released, I can confirm that cuda 11.0 will work on ubuntu 20.04, with tensorflow. Its just there will be nasty error about:

Your CUDA software stack is old. We fallback to the NVIDIA driver for some compilation. Update your CUDA version to get the best performance. The ptxas error was: ptxas fatal   : Value 'sm_86' is not defined for option 'gpu-name'

But it do not affect training/inferencing
